flatcar-linux.json

```diff
@@ -0,0 +1,63 @@
+{
+    "variables": {
+        "flatcar_channel": "alpha",
+        "flatcar_version": "current",
+        "iso_checksum": "",
+        "iso_checksum_type": "none",
+        "disk_size": "40000",
+        "memory": "2048M",
+        "boot_wait": "45s",
+        "headless": "false"
+    },
+
+    "builders": [
+        {
+            "type": "qemu",
+            "iso_url": "https://{{user `flatcar_channel` }}.release.flatcar-linux.net/amd64-usr/current/flatcar_production_iso_image.iso",
+            "iso_checksum": "{{ user `iso_checksum` }}",
+            "iso_checksum_type": "{{ user `iso_checksum_type` }}",
+            "output_directory": "builds",
+            "shutdown_command": "sudo shutdown now",
+            "disk_size": "{{ user `disk_size` }}",
+            "format": "qcow2",
+            "accelerator": "kvm",
+            "vm_name": "flatcar-linux-{{user `flatcar_channel` }}.qcow2",
+            "boot_wait": "{{user `boot_wait` }}",
+            "ssh_username": "core",
+            "ssh_password": "packer",
+            "headless": "{{ user `headless` }}",
+            "boot_command": [
+                "sudo passwd core<enter><wait>",
+                "packer<enter>",
+                "packer<enter>",
+                "sudo systemctl start sshd.service<enter>"
+            ],
+            "qemuargs": [
+                [ "-m", "{{user `memory` }}" ]
+            ]
+        }
+    ],
+
+    "provisioners": [
+        {
+            "type": "file",
+            "source": "ignition.json",
+            "destination": "/tmp/ignition.json"
+        },
+        {
+            "type": "shell",
+            "inline": [
+                "sudo flatcar-install -d /dev/vda -C {{ user `flatcar_channel` }} -i /tmp/ignition.json"
+            ]
+        }
+    ],
+
+    "post-processors": [
+        {
+            "type": "vagrant",
+            "keep_input_artifact": true,
+            "output": "builds/flatcar-{{user `flatcar_channel`}}-{{user `flatcar_version`}}-{{.Provider}}.box",
+            "vagrantfile_template": "Vagrantfile"
```

Comment:
I couldn't find documentation on what this template is for?

Reply:
Do we need to include the `Vagrantfile` in the repo at all?

Reply:
With it included, it prevents people from doing e.g. `vagrant init user/flatcar-alpha` after building and adding the box, because a Vagrantfile exists already..

Reply:
Maybe it is just there for ease of testing? @dongsupark what do you think?

Reply:
@schu according to the docs https://www.packer.io/docs/post-processors/vagrant.html#vagrantfile_template, It is a `Path to a template to use for the Vagrantfile that is packaged with the box.` I do not really understand much of it/why at the moment.

Reply:
Let's move `Vagrantfile` to a new subdirectory `workspace` or so, to isolate the runtime environment from the build environment.

I'm not sure about the `vagrantfile_template`. It's clear that setting this variable, the specified Vagrantfile is included in the box image. Though I don't know why that should be included. If it works fine without `vagrantfile_template`, I'm fine with removing the option.

Reply:
I tested this again and the values seem to be used from the template where not specified in the user Vagrantfile (e.g. memory). So I think there should be a "template" but it should only contain the defaults we want to set for the box build (e.g. provider libvirt) and no name.

```
Vagrant.configure('2') do |config|
  config.ssh.username = 'core'
  config.ssh.insert_key = true
  config.vm.synced_folder '.', '/vagrant', disabled: true
  config.vm.provider :libvirt do |libvirt|
    libvirt.cpus = 2
    libvirt.memory = 2048
    libvirt.driver = 'kvm'
    libvirt.host = ''
    libvirt.connect_via_ssh = false
    libvirt.storage_pool_name = 'default'
  end
end
```

Reply:
Or maybe we should take it out completely, and let packer build with its own defaults.

Reply:
I think it's better we set them, e.g. to avoid a problem due to not enough memory.

Reply:
Actually, I think it would fail w/o since the default user is not `core`